lib/AST/ASTContext.cpp

```diff
@@ -11182,3 +11182,46 @@ APFixedPoint ASTContext::getFixedPointMin(QualType Ty) const {
   assert(Ty->isFixedPointType());
   return APFixedPoint::getMin(getFixedPointSemantics(Ty));
 }
+
+RecordDecl *ASTContext::getCachedTypeApp(const RecordDecl *Base, ArrayRef<const Type *> TypeArgs) {
+  auto it = CachedTypeApps.find(std::make_pair(Base, TypeArgs));
```

Comment:
Nit: Can we use: "const auto" for the iterator? Similar observation in the methods below.

Reply:
Done